services/consensusalgo/leanhelixconsensus/service.go

```diff
@@ -58,62 +59,45 @@ func newMetrics(m metric.Factory, consensusTimeout time.Duration) *metrics {
 	}
 }
 
-type BlockPairWrapper struct {
-	blockPair *protocol.BlockPairContainer
-}
-
-func (b *BlockPairWrapper) Height() lhprimitives.BlockHeight {
-	return lhprimitives.BlockHeight(b.blockPair.TransactionsBlock.Header.BlockHeight())
-}
-
-func (b *BlockPairWrapper) BlockHash() lhprimitives.Uint256 {
-	// TODO This is surely incorrect, fix to use the right hash
-	return lhprimitives.Uint256(b.blockPair.TransactionsBlock.Header.MetadataHash())
-}
-
-func NewBlockPairWrapper(blockPair *protocol.BlockPairContainer) *BlockPairWrapper {
-	return &BlockPairWrapper{
-		blockPair: blockPair,
-	}
-}
-
 func NewLeanHelixConsensusAlgo(
 	ctx context.Context,
 	gossip gossiptopics.LeanHelix,
 	blockStorage services.BlockStorage,
+
 	consensusContext services.ConsensusContext,
 	logger log.BasicLogger,
 	config Config,
 	metricFactory metric.Factory,
 
 ) services.ConsensusAlgoLeanHelix {
 
+	comm := NewNetworkCommunication(gossip)
+	mgr := NewKeyManager(config.NodePublicKey(), config.NodePrivateKey())
+	provider := NewBlockProvider(config.LeanHelixConsensusRoundTimeoutInterval(), config.NodePublicKey(), config.NodePrivateKey())
 	electionTrigger := leanhelix.NewTimerBasedElectionTrigger(config.LeanHelixConsensusRoundTimeoutInterval())
 
 	s := &service{
-		gossip:                  gossip,
-		blockStorage:            blockStorage,
-		consensusContext:        consensusContext,
-		logger:                  logger.WithTags(LogTag),
-		config:                  config,
-		metrics:                 newMetrics(metricFactory, config.LeanHelixConsensusRoundTimeoutInterval()),
-		leanHelix:               nil,
-		messageReceivers:        make(map[int]func(ctx context.Context, message leanhelix.ConsensusRawMessage)),
-		messageReceiversCounter: 0,
+		blockStorage:     blockStorage,
+		comm:             comm,
+		consensusContext: consensusContext,
+		logger:           logger.WithTags(LogTag),
+		config:           config,
+		metrics:          newMetrics(metricFactory, config.LeanHelixConsensusRoundTimeoutInterval()),
+		leanHelix:        nil,
```

Comment:
why are you explicitly setting it as nil?

Reply:
Refactored so not necessary to set to nil. 
Originally wanted to emphasize that this property needed to be set later on.